File: app/src/main/java/com/osrsd/cache/def/SequenceDefinition.java
package com.osrsd.cache.def;

import com.osrsd.spring.domain.Definition;
import lombok.Data;

@Data
public final class SequenceDefinition implements Definition {

    private int id;
    private int[] frameIDs;
    private int[] chatFrameIds;
    private int[] frameLengths;
    private int[] frameSounds;
    private int frameStep;
    private int[] interleaveLeave;
    private boolean stretches;
    private int forcedPriority;
    private int leftHandItem;
    private int rightHandItem;
    private int maxLoops;
    private int precedenceAnimating;
    private int priority;
    private int replyMode;

    public SequenceDefinition(int id) {
        this.id = id;
        frameStep = -1;
        stretches = false;
        forcedPriority = 5;
        leftHandItem = -1;
        rightHandItem = -1;
        maxLoops = 99;
        precedenceAnimating = -1;
        priority = -1;
        replyMode = 2;
    }

}

File: app/src/main/kotlin/com/osedit/spring/domain/SequenceDefinition.kt
package com.osedit.spring.domain

import com.osedit.cache.Config
import com.osedit.cache.Indices
import com.osedit.cache.provider.SequenceProvider
import java.nio.ByteBuffer
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class SequenceDefinition(@Id private val id: Int? = 0) : Definition {
    @Column(length = 1191)
    var frameIDs: IntArray? = null
    @Column(length = 387)
    var chatFrameIds: IntArray? = null
    @Column(length = 1191)
    var frameLengths: IntArray? = null
    @Column(length = 523)
    var frameSounds: IntArray? = null
    var frameStep: Int = -1
    var interleaveLeave: IntArray? = null
    var stretches: Boolean = false
    var forcedPriority: Int = 5
    var leftHandItem: Int = -1
    var rightHandItem: Int = -1
    var maxLoops: Int = 99
    var precedenceAnimating: Int = -1
    var priority: Int = -1
    var replyMode: Int = 2

    override fun path(): String {
        return "/sequences"
    }

    override fun config(): Config {
        return Config.SEQUENCES
    }

    override fun indices(): Indices {
        return Indices.CONFIG
    }

    override fun decode(byteBuffer: ByteBuffer): Definition {
        return SequenceProvider.decode(byteBuffer, this)
    }
}